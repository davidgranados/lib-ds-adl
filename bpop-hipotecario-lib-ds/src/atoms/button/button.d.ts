import { DefaultButtonProps } from "../../types/button";
import { ButtonBaseProps } from "../button-base";
import { HTMLPropsFor } from "../../types";

export type ButtonSize = "xs" | "sm" | "md" | "lg";

export interface ButtonProps extends DefaultButtonProps, ButtonBaseProps, HTMLPropsFor<"button"> {
  size?: ButtonSize;
}
