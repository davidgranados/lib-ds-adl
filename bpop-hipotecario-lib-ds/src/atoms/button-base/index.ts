export { default as ButtonBase } from "./button-base";

export type { ButtonBaseProps, ButtonFontSize } from "./button-base.d";
