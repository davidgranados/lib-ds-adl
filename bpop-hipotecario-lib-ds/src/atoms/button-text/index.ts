export { default as ButtonText } from "./button-text";

export { validSizes, validVariants } from "./helpers";
