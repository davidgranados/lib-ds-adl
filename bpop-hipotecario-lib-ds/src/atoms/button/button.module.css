.default {
  border: solid var(--border-width-button) var(--color-base-transparent);
  border-radius: var(--border-radius-button);
}

.primary {
  background-color: var(--background-color-button-primary-enabled);
  color: var(--font-color-button-primary-enabled);
}

.primary--focus {
  border: solid var(--border-width-button) var(--border-color-button-primary);
  background-color: var(--background-color-button-primary-focus);
}

.primary--hover {
  background-color: var(--background-color-button-primary-hover);
  color: var(--font-color-button-primary-hover);
}

.primary--active {
  background-color: var(--background-color-button-primary-active);
}

.primary:disabled {
  background-color: var(--background-color-button-primary-disabled);
  color: var(--font-color-button-primary-disabled);
}

.secondary {
  background-color: var(--background-color-button-secondary-enabled);
  color: var(--font-color-button-secondary-enabled);
}

.secondary--focus {
  border: solid var(--border-width-button) var(--border-color-button-secondary);
  background-color: var(--background-color-button-secondary-focus);
}

.secondary--hover {
  background-color: var(--background-color-button-secondary-hover);
  color: var(--font-color-button-secondary-hover);
}

.secondary--active {
  background-color: var(--background-color-button-secondary-active);
}

.secondary:disabled {
  background-color: var(--background-color-button-secondary-disabled);
  color: var(--font-color-button-secondary-disabled);
}

.size-xs {
  height: 32px;
  padding: 8px 16px;
}

.size-sm {
  height: 40px;
  padding: 8px 24px;
}

.size-md {
  height: 48px;
  padding: 12px 32px;
}

.size-lg {
  height: 56px;
  padding: 16px 32px;
}
