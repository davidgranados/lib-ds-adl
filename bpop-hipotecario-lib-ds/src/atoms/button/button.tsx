import React from "react";
import classNames from "classnames";

import { ButtonProps } from "../../types";

import buttonStyles from "./button.module.css";

export const Button: React.FC<ButtonProps> = ({
  label,
  type = "button",
  variant = "primary",
  size = "md",
  ...props
}) => {
  const buttonClassName = classNames(buttonStyles["default"], buttonStyles[variant], buttonStyles[`size-${size}`]);
  return (
    <button type={type} className={buttonClassName} {...props}>
      {label}
    </button>
  );
};
