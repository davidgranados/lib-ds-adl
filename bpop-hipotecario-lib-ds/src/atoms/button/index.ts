export { default as Button } from "./button";

export { validSizes, validVariants } from "./helpers";
