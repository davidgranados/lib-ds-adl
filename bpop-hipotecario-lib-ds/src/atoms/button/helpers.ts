import { ButtonVariant } from "../../types";
import { ButtonSize } from "./index";
import { ButtonFontSize } from "../button-base";

export const validSizes: ButtonSize[] = ["xs", "sm", "md", "lg"];
export const validVariants: ButtonVariant[] = ["primary", "secondary"];
export const fontSizeMapping: Record<ButtonSize, ButtonFontSize> = {
  xs: "sm",
  sm: "md",
  md: "md",
  lg: "lg",
};
