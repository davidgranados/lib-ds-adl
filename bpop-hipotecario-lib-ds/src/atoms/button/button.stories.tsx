/* istanbul ignore file */
import React from "react";
import { ComponentMeta, ComponentStory } from "@storybook/react";

import Button from "./button";
import { validSizes, validVariants } from "./helpers";

export default {
  title: "Atoms/Button",
  component: Button,
  args: {
    variant: "primary",
    disabled: false,
    text: "Button",
  },
  argTypes: {
    variant: { options: validVariants, control: "select" },
    size: { options: validSizes, control: "select" },
    disabled: { control: "boolean" },
    onFocus: { action: "focused" },
    onBlur: { action: "blurred" },
    onClick: { action: "clicked" },
    onMouseDown: { action: "mouse down" },
    onMouseUp: { action: "mouse up" },
    onMouseEnter: { action: "mouse enter" },
    onMouseLeave: { action: "mouse leave" },
    onTouchStart: { action: "touch start" },
    onTouchEnd: { action: "touch end" },
  },
} as ComponentMeta<typeof Button>;

const Template: ComponentStory<typeof Button> = (args) => <Button {...args} />;

export const Default = Template.bind({});
