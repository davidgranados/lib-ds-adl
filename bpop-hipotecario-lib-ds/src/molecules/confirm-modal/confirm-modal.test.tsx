import React from "react";
import { cleanup } from "@testing-library/react";
import { shallow } from "enzyme";

import ConfirmModal from "./confirm-modal";
import { Button } from "../../atoms/button";

afterEach(cleanup);
describe("ConfirmModal", () => {
  let show = false;
  let isMainButtonClicked = false;
  let isSecondaryButtonClicked = false;
  const component = shallow(
    <ConfirmModal
      id={"confirm-modal"}
      show={show}
      title={"Modal title"}
      mainButtonText={"main"}
      secondaryButtonText={"second"}
      onMainButtonClick={() => {
        isMainButtonClicked = true;
      }}
      onSecondaryButtonClick={() => {
        isSecondaryButtonClicked = true;
      }}
      onToggle={(s) => {
        show = s;
      }}
    />
  );
  test("ConfirmModal should renders ok", () => {
    expect(component).toMatchSnapshot();
  });

  test("ConfirmModal should click on main button ok", async () => {
    const expected = true;
    component.find(Button).simulate("click");
    expect(isMainButtonClicked).toBe(expected);
  });

  test("ConfirmModal should click on secondary button ok", () => {
    const expected = true;
    component.find("#confirm-modal-secondary-button").simulate("click");
    expect(isSecondaryButtonClicked).toBe(expected);
  });
});
