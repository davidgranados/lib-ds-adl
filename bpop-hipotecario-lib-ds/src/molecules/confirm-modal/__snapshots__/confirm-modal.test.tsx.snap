// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmModal ConfirmModal should renders ok 1`] = `
<Dialog
  dataTestId="confirm-modal-dialog"
  id="confirm-modal-dialog"
  onToggle={[Function]}
  show={false}
  width={320}
>
  <div
    className="confirm-modal-wrapper"
    data-testid="confirm-modal-wrapper"
    id="confirm-modal-wrapper"
  >
    <div
      className="img"
    >
      <span
        className="icon icon-exclamation-open-circle"
      />
    </div>
    <div
      className="title"
    >
      Modal title
    </div>
    <div
      className="description"
    />
    <div
      className="button"
    >
      <Button
        dataTestId="confirm-modal-primary-button"
        fullWidth={true}
        id="confirm-modal-primary-button"
        onClick={[Function]}
        size="md"
        text="main"
        variant="primary"
      />
    </div>
    <span
      className="secondary-button"
      data-testid="confirm-modal-secondary-button"
      id="confirm-modal-secondary-button"
      onClick={[Function]}
    >
      second
    </span>
  </div>
</Dialog>
`;
